src/coreclr/src/gc/env/gcenv.object.h

```diff
@@ -1,193 +1,10 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-
 #ifndef __GCENV_OBJECT_H__
-#define __GCENV_OBJECT_H__
+#define __GCENV_OJECT_H__
 
-// ARM requires that 64-bit primitive types are aligned at 64-bit boundaries for interlocked-like operations.
-// Additionally the platform ABI requires these types and composite type containing them to be similarly
-// aligned when passed as arguments.
-#ifdef TARGET_ARM
-#define FEATURE_64BIT_ALIGNMENT
+#ifdef HAVE_CORE_GC
```

Comment:
This should not be needed here. You should be able to provide Mono-specific `gcenv.h` that has the right environment definition for Mono.

This file is not actually used by the regular CoreCLR GC build. The regular CoreCLR build has its own definition of the environment in https://github.com/dotnet/runtime/blob/master/src/coreclr/src/vm/gcenv.h.

Reply:
Thanks. I am trying to work out these details.